numba/core/typed_passes.py

```diff
@@ -69,13 +69,14 @@ def type_inference_stage(typingctx, interp, args, return_type, locals={},
             infer.seed_type(k, v)
 
         infer.build_constraint()
-        infer.propagate(raise_errors=raise_errors)
+        # return errors in case of partial typing
+        errs = infer.propagate(raise_errors=raise_errors)
         typemap, restype, calltypes = infer.unify(raise_errors=raise_errors)
 
     # Output all Numba warnings
     warnings.flush()
 
-    return typemap, restype, calltypes
+    return typemap, restype, calltypes, errs
```

Comment:
With 4 elements returning, i think it'd be clearer (esp for the caller) if this return a namedtuple instead to avoid a unnamed unpacking and the `_, _, _`.

Reply:
I agree. Done.

Reply:
Should `type_inference_stage` also return the same? It's use is more prevalent and also leads to the same `x, _, _, _` pattern.

Reply:
@stuartarchibald `type_inference_stage` is returning a namedtuple in this PR now. What do you mean?